### ISSUE TITLE
HTML minify slow in generate

### ISSUE THREAD

**Issue by jroxendal** [nuxt/nuxt.js#746]
I find that minifying the HTML in generate takes around 100ms for each rendered route, which seems unreasonably slow: my ~400 routes generate at 70 seconds but if I deactivate minify I shave off a full 40 seconds! Any idea on how to speed up minify?

**Comment by ausir0726** [comment]
Unrestrained idea :

give up node.js html minify .

after you build , run a golang html minify lib .
( don't know it will be faster ? )

**Comment by jroxendal** [comment]
It certainly aught to be faster. Good input, anyone else? What about the
cluster module?

**Comment by ausir0726** [comment]
@jroxendal 
honest , nuxt.js just use html-minify package , If you want enhance html-minify , it should be report on html-minify repo . Suggest it use muti-thread minify , or find another more effectiveness html minify packeage , let developer check it will be better for nuxt.js .

just my personal opinion .

If you want maximum speed , use go / c bin file to minify html file maybe can get fastest processing time .
just modiy your package "generate":"nuxt generate" to "nuxt generate &&  ./minify dist"
try it . I don't have 400+ html in project , so , maybe you can try this solution .

**Comment by ausir0726** [comment]
as my personal interest
I tried benchmark about default html-minifier and golang html minify.
## result
|                     | html-minifier 400 | go minify 400 | html-minifier 4000 | go minify 4000 |
|---------------------|-------------------|---------------|--------------------|----------------|
| generate each file  | 120ms             | 25ms          | 120ms              | 25ms           |
| total generate time | 79.87s            | 53.28s        | 627.32s            | 327.20s        |
| minify time         | ≒ 80ms * 400 = 16s  | 0.71s         | ≒80ms * 4000 = 320s  | 7.37s          |
| save                |                   | 26.59s        |                    | 300.12s        |

seems you can use go html minify to save you generate time .
@jroxendal  for your reference .

## TL;DR

I just copy my index.vue to minify-test/_test.vue
and generate 400 and 4000 routes generate .
first generate with default html-minifier setting , second generate is set all html-minifier setting to false , after generated run go html minify command all dist folder `minify.exe -o dist/ -r dist`

#### first test 

400 routes with default setting
![screenshot_7](https://cloud.githubusercontent.com/assets/7247064/26282778/ab99d794-3e4a-11e7-9d80-425fee7dfb02.png)

400 routes with go minify
![screenshot_10](https://cloud.githubusercontent.com/assets/7247064/26282790/d18d0282-3e4a-11e7-8fff-7cd77b04a27d.png)

#### second test

4000 routes with default setting
![4000-default](https://cloud.githubusercontent.com/assets/7247064/26282793/e208f3dc-3e4a-11e7-8a09-a4cb2753024d.png)

4000 routes with go minify
![4000-go](https://cloud.githubusercontent.com/assets/7247064/26282795/e9fc9404-3e4a-11e7-9ca2-f830db33fc00.png)

test computer
![screenshot_11](https://cloud.githubusercontent.com/assets/7247064/26282803/11867e22-3e4b-11e7-822e-3376d02d499f.png)

**Comment by ausir0726** [comment]
@Atinux 
seems like no config can just skip html-minifier , if can just skip html-minifier maybe can save more generate time . just a suggest for features ( maybe can wait some one PR 😆 )

**Comment by Atinux** [comment]
@ausir0729 I really love your answers! Really great content inside!

I'm adding the option to disable the `html-minifier` by setting `generate.minify: false` in `nuxt.config.js`. No need for a PR :)

**Comment by ausir0726** [comment]
@Atinux  I already done add `generate.minify : false `
and make test report .
I will PR few mins later .

**Comment by Atinux** [comment]
Please PR into `dev` branch @ausir0726

**Comment by ausir0726** [comment]
forget the PR , you already finish :D nothing better .

**Comment by jroxendal** [comment]
This was great, thanks @ausir0726!. I have a PR on the way that improves render times for dynamic routes, combined with this even thousands of routes should generate and minify quickly 👍
@atinux you want all PRs to dev?

**Comment by Atinux** [comment]
From now on, all PR should be on `dev` yes :)

**Comment by vkatsar** [comment]
@ausir0726 is this works for nuxt express template?
